pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeConsumptionRateManager.java

```diff
@@ -126,14 +137,17 @@ public interface ConsumptionRateLimiter {
   static class RateLimiterImpl implements ConsumptionRateLimiter {
     private final double _rate;
     private final RateLimiter _rateLimiter;
+    private MetricEmitter _metricEmitter;
 
-    private RateLimiterImpl(double rate) {
+    private RateLimiterImpl(double rate, MetricEmitter metricEmitter) {
       _rate = rate;
       _rateLimiter = RateLimiter.create(rate);
+      _metricEmitter = metricEmitter;
     }
 
     @Override
     public void throttle(int numMsgs) {
+      _metricEmitter.emitMetric(numMsgs, _rate, Clock.systemUTC().instant());
```

Comment:
@sajjad-moradi, what do you think about only emitting this metric when `InstanceHolder.INSTANCE._isThrottlingAllowed`? I notice this keeps rocketing >100% whenever servers are starting and haven't passed status checks and rate limiting is not yet enabled.